tickets/src/components/content/NewEventContent.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { Component } from 'react'
+import Head from 'next/head'
+import { RouterState } from 'connected-react-router'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+import { lockRoute } from '../../utils/routes'
+import Layout from '../interface/Layout'
+import withConfig from '../../utils/withConfig'
+import { pageTitle } from '../../constants'
+import EventInfo from '../interface/EventInfo'
+import EventLinks from '../interface/EventLinks'
+import EventDescription from '../interface/EventDescription'
+import PurchaseTicket from '../interface/PurchaseTicket'
+import EventTicket from '../interface/EventTicket'
+import Media from '../../theme/media'
+
+interface UnlockWindow extends Window {
+  unlockProtocol?: any
+  unlockProtocolConfig?: any
+}
+
+declare var window: UnlockWindow | undefined
+
+enum PaywallStatus {
+  Locked = 'locked',
+  Unlocked = 'unlocked',
+  Unknown = 'unknown',
+}
+
+interface EventContentProps {
+  lockAddress: string
+  config: { env: string }
+  event: any
+}
+
+interface EventContentState {
+  paywallStatus: PaywallStatus
+}
+
+export class EventContent extends Component<
+  EventContentProps,
+  EventContentState
+> {
+  constructor(props: EventContentProps) {
+    super(props)
+
+    const { lockAddress } = props
+
+    this.state = {
+      paywallStatus: PaywallStatus.Unknown,
+    }
+
+    if (typeof window !== 'undefined') {
+      window.addEventListener('unlockProtocol', this.setPaywallStatus)
+
+      // TODO: Make this nicer
+      window.unlockProtocolConfig = {
+        persistentCheckout: false,
+        locks: {
+          [lockAddress]: {},
+        },
+        callToAction: {
+          default: 'Buy a ticket',
+        },
+      }
+    }
+  }
+
+  componentWillUnmount = () => {
+    if (window) {
+      window.removeEventListener('unlockProtocol', this.setPaywallStatus)
+    }
+  }
+
+  setPaywallStatus = (event: Event) => {
+    const d = (event as any).detail
+
+    // Making sure we only set state to expected values, since it's possible for
+    // someone to manually send a CustomEvent
+    if (d === PaywallStatus.Locked || d === PaywallStatus.Unlocked) {
+      this.setState({
+        paywallStatus: d,
+      })
+    }
+  }
+
+  paywallScriptUrl = (): string => {
+    const {
+      config: { env },
+    } = this.props
+    const extension = '/static/unlock.1.0.min.js'
+
+    // TODO: add test url?
+    const baseUrls: { [key: string]: string } = {
+      dev: 'http://localhost:3001',
```

Comment:
Should these be in here? I am not  in the loop on how the application is written but could we pass these in from elsewhere?

Reply:
Good question, I'm not 100% sure how we handle this elsewhere, but I'll look into it

Reply:
huh, this clearly does not belong there :(